src/AWS.Deploy.CLI/Commands/GenerateDeploymentProjectCommand.cs

```diff
@@ -0,0 +1,218 @@
+                    "Please provide an empty directory path and try again." + Environment.NewLine;
+            
+            if (_directoryManager.ExistsInsideDirectory(targetApplicationDirectoryFullPath, saveCdkDirectoryPath))
+                errorMessage += "The directory used to save the CDK deployment project is contained inside of " +
+                    "the target application project directory. Please specify a different directory and try again.";
+
+            return errorMessage.Trim();
+        }
+
+        /// <summary>
+        /// Checks if the location of the saved CDK deployment project is monitored by a source control system.
+        /// </summary>
+        /// <param name="saveCdkDirectoryPath">Relative or absolute path of the directory at which the CDK deployment project will be saved.</param>
+        /// <returns></returns>
+        private async Task<bool> IsDirectoryUnderSourceControl(string saveCdkDirectoryPath)
+        {
+            var gitStatusResult = await _commandLineWrapper.TryRunWithResult("git status", saveCdkDirectoryPath);
+            var svnStatusResult = await _commandLineWrapper.TryRunWithResult("svn status", saveCdkDirectoryPath);
+            return gitStatusResult.Success || svnStatusResult.Success;
+        }
+
+        /// <summary>
+        /// Generates a snapshot of the deployment recipe inside the location at which the CDK deployment project is saved.
+        /// </summary>
+        /// <param name="recommendation"><see cref="Recommendation"/></param>
+        /// <param name="saveCdkDirectoryPath">Relative or absolute path of the directory at which the CDK deployment project will be saved.</param>
+        /// <param name="projectDisplayName">The name of the deployment project that will be displayed in the list of available deployment options.</param>
+        private async Task GenerateDeploymentRecipeSnapShot(Recommendation recommendation, string saveCdkDirectoryPath, string projectDisplayName)
+        {
+            var targetApplicationDirectoryName = _directoryManager.GetDirectoryInfo(_targetApplicationFullPath).Parent.Name;
+            var recipeSnapshotFileName = _directoryManager.GetDirectoryInfo(saveCdkDirectoryPath).Name + ".recipe";
+            var recipeSnapshotFilePath = Path.Combine(saveCdkDirectoryPath, recipeSnapshotFileName);
+
+            var recipeBody = await _fileManager.ReadAllTextAsync(recommendation.Recipe.RecipePath ?? "");
+            var recipe = JsonConvert.DeserializeObject<RecipeDefinition>(recipeBody);
+            
+            var recipeName = string.IsNullOrEmpty(projectDisplayName) ?
+                $"Deployment project for {targetApplicationDirectoryName} to {recommendation.Recipe.TargetService}"
+                : projectDisplayName;
+
+            recipe.Id = Guid.NewGuid().ToString();
+            recipe.Name = recipeName;
+            recipe.CdkProjectTemplateId = string.Empty;
+            recipe.CdkProjectTemplate = string.Empty;
```

Comment:
should we consider making these nullable?

Reply:
Made them nullable